ckanext/sse/plugin.py

```
@@ -268,9 +270,25 @@ def get_signal_subscriptions(self):
 
     # IResourceController
+    def before_resource_create(self, context, resource):
+        # SI-7: allowlist + checksum before the file reaches storage.
+        upload_security.before_create(context, resource)
+
+    def before_resource_update(self, context, current, resource):
+        upload_security.before_update(context, current, resource)
+
     def after_resource_create(self, context, data_dict):
+        upload_security.after_change(context, data_dict)
         id = data_dict.get("id")
```

Ruff (0.16.2)

[error] Variable `id` is shadowing a Python builtin

(A001)